backend/src/test/asegroup1/api/models/landregistry/LandRegistryQueryConstraintTest.java

```diff
@@ -0,0 +1,321 @@
+package asegroup1.api.models.landregistry;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.Random;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import asegroup1.api.models.landregistry.LandRegistryQueryConstraint.PropertyType;
+import asegroup1.api.models.landregistry.LandRegistryQueryConstraint.RangeConstraint;
+import asegroup1.api.models.landregistry.LandRegistryQuerySelect.Selectable;
+
+/**
+ * @author Richousrick
+ *
+ */
+class LandRegistryQueryConstraintTest {
+
+	LandRegistryQueryConstraint constraint;
+
+	private static String[] genRandomPostcodes() {
+		return new String[] { "OX14 1WH", "L18 9SN", "TN27 8JG", "PL8 2EE" };
```

Comment:
Maybe call this `getRandomPostCodes` rather than gen, as the postcodes are not generated on the method call